mlflow/sagemaker/container/__init__.py

```diff
@@ -57,10 +86,39 @@ def _serve():
 
     Read the MLmodel config, initialize the Conda environment if needed and start python server.
     """
-    m = Model.load("/opt/ml/model/MLmodel")
-    if pyfunc.FLAVOR_NAME not in m.flavors:
-        raise Exception("Only supports pyfunc models and this is not one.")
-    conf = m.flavors[pyfunc.FLAVOR_NAME]
+    model_config_path = os.path.join(MODEL_PATH, "MLmodel")
+    m = Model.load(model_config_path)
+
+    if ENV_KEY_DEPLOYMENT_CONFIG in os.environ:
+        deployment_config = yaml.load(os.environ[ENV_KEY_DEPLOYMENT_CONFIG])
```

Comment:
I would say lets figure out the flavor outside of the container and ALWAYS set the environment variable. Less code paths to worry about.

Reply:
I'm fine with that, as long as we recognize that users must have the latest version of MLflow to use the latest version of the container. Deploying from an older version of MLflow with a URL for a container built against the latest version will fail. Is that okay with you?

Reply:
I think that is ok.

Reply:
We can also default to pyfunc on the container side. That way the old behavior is preserved for older clients.